frontend/app/services/SalesforceService.scala

```diff
@@ -51,21 +56,41 @@ class SalesforceService(salesforceConfig: SalesforceConfig) extends api.Salesfor
     )
   }.getOrElse(Json.obj())
 
-  private def initialData(user: IdUser, formData: JoinForm): JsObject = {
-    Seq(Json.obj(
-      Keys.EMAIL -> user.primaryEmailAddress,
-      Keys.FIRST_NAME -> formData.name.first,
-      Keys.LAST_NAME -> formData.name.last,
-      Keys.MAILING_STREET -> formData.deliveryAddress.line,
-      Keys.MAILING_CITY -> formData.deliveryAddress.town,
-      Keys.MAILING_STATE -> formData.deliveryAddress.countyOrState,
-      Keys.MAILING_POSTCODE -> formData.deliveryAddress.postCode,
-      Keys.MAILING_COUNTRY -> formData.deliveryAddress.country.getOrElse(UK).alpha2,
-      Keys.ALLOW_MEMBERSHIP_MAIL -> true
-    )) ++ Map(
-      Keys.ALLOW_THIRD_PARTY_EMAIL -> formData.marketingChoices.thirdParty,
-      Keys.ALLOW_GU_RELATED_MAIL -> formData.marketingChoices.gnm
-    ).collect { case (k, Some(v)) => Json.obj(k -> v) }
+  private def contributorData(customerOpt: Option[Customer]): JsObject =
```

Comment:
Looking more carefully, I can see that we write this information for membership too (although I think the tier update is the important part here). Generally speaking payment data only belongs in Zuora - I don't understand why we are writing it into Salesforce as well